core/state_update.go

```diff
@@ -51,6 +51,7 @@ func (d *StateDiff) Commitment() *felt.Felt {
 	for addr := range d.ReplacedClasses {
 		deployedReplacedAddresses = append(deployedReplacedAddresses, addr)
 	}
+	// this will insert number_of_deployed_contracts into slice which then will be moved to the beginning by this sorting, right?
```

Comment:
nope `hashOfDeployedContracts.Update` straight up updates the digest that we are calculating here.